compiler/src/main/scala/org/scalajs/core/compiler/PrepJSInterop.scala

```diff
@@ -693,7 +694,7 @@ abstract class PrepJSInterop extends plugins.PluginComponent
         }
 
         val ownerLoadSpec = jsInterop.jsNativeLoadSpecOf(sym.owner)
-        val jsName = jsInterop.jsNameOf(sym)
+        val JSName.Literal(jsName) = jsInterop.jsNameOf(sym)
```

Comment:
Ah! But those IR changes also require to change the `Analyzer` in a non-obvious way. Because reaching a `classOf[Foo.Bar]` requires to reach the contents of the `JSName.Computed`s in the load spec of `Foo.Bar`. And that requires more infos, and that's non trivial.

At this point I'm inclined to disallow `@JSName(sym)` on native `object`s and `class`es in native `object`s, as an implementation restriction. It's less critical than the rest, and obviously we need more time to get a good grasp of what to do here.